packages/react-core/src/hooks/__tests__/useIsSignedIn.test.ts

```diff
@@ -0,0 +1,90 @@
+import useIsSignedIn from '../useIsSignedIn';
+import { act, renderHook } from '@testing-library/react-hooks';
+import { getCurrentUser } from 'aws-amplify/auth';
+import { GetCurrentUserOutput } from 'aws-amplify/auth';
+import { Hub } from 'aws-amplify/utils';
+
+jest.mock('aws-amplify/auth', () => ({
+  getCurrentUser: jest.fn(),
+}));
+
+describe('useIsSignedIn', () => {
+  it('should be falsy if user is not signed in', () => {
+    const mockError = new Error('Authorization error');
+    (getCurrentUser as jest.Mock).mockRejectedValueOnce(mockError);
+
+    const { result } = renderHook(() => useIsSignedIn());
+    expect(result.current).toBeFalsy();
```

Comment:
Prefer explicit `.toBe(true)` or `.toBe(false)` over `toBeTruthy()`/`toBeFalsy()`. Falsy/truthy assertions pass against less strict values, for example `null` for falsy